src/back/spv/index.rs

```diff
@@ -42,32 +42,81 @@ impl<'w> BlockContext<'w> {
         array: Handle<crate::Expression>,
         block: &mut Block,
     ) -> Result<Word, Error> {
-        // Naga IR permits runtime-sized arrays as global variables or as the
-        // final member of a struct that is a global variable. SPIR-V permits
-        // only the latter, so this back end wraps bare runtime-sized arrays
-        // in a made-up struct; see `helpers::global_needs_wrapper` and its uses.
-        // This code must handle both cases.
-        let (structure_id, last_member_index) = match self.ir_function.expressions[array] {
+        // Naga IR permits runtime-sized arrays as global variables, or as the
+        // final member of a struct that is a global variable, or one of these
+        // inside a buffer that is itself an element in a buffer bindings array.
+        // SPIR-V requires that runtime-sized arrays are wrapped in structs.
+        // See `helpers::global_needs_wrapper` and its uses.
+        let (opt_array_index, global_handle, opt_last_member_index) = match self
+            .ir_function
+            .expressions[array]
+        {
+            // Note that SPIR-V forbids `OpArrayLength` on a variable pointer,
+            // so we aren't handling `crate::Expression::Access` here.
```

Comment:
This is true, but irrelevant. `OpAccessChain` applied to a `OpVariable` doesn't produce a variable pointer. There should be no reason we can't support `Access` here. The following validates just fine:

```
                   OpCapability Shader
                   OpCapability Linkage
                   OpExtension "SPV_KHR_storage_buffer_storage_class"
                   OpMemoryModel Logical GLSL450

                   OpDecorate %struct Block
                   OpMemberDecorate %struct 0 Offset 0
                   OpMemberDecorate %struct 1 Offset 4

            %u32 = OpTypeInt 32 0
        %ptr_u32 = OpTypePointer StorageBuffer %u32
  %runtime_array = OpTypeRuntimeArray %u32
         %struct = OpTypeStruct %u32 %runtime_array
  %binding_array = OpTypeRuntimeArray %struct
         %ptr_ba = OpTypePointer StorageBuffer %binding_array

            %one = OpConstant %u32 1
                   
         %global = OpVariable %ptr_ba StorageBuffer

      %func_type = OpTypeFunction %u32 %u32 %u32
           %func = OpFunction %u32 None %func_type
         %index1 = OpFunctionParameter %u32
         %index2 = OpFunctionParameter %u32
       %prologue = OpLabel
            %ptr = OpAccessChain %ptr_u32 %global %index1 %one %index2
          %value = OpLoad %u32 %ptr
                   OpReturnValue %value
                   OpFunctionEnd
```

Reply:
Great point!
Impl is adjusted accordingly now.